internal/platform/platform.go

```diff
@@ -0,0 +1,387 @@
+	Tempsa2         float32 `json:"tcoTemp_sa2"`
+	Tempsc10pzPanel float32 `json:"tcoTemp_sc10_pzPanel"`
+	Tempstr         float32 `json:"tcoTemp_str"`
+}
+
+// AttitudeRecord for variables in group: "PreciseAttitudeEstimation"
+type AttitudeRecord struct {
+	Time                time.Time
+	AttitudeState       [4]float64    `json:"afsAttitudeState"`
+	AttitudeUncertainty [3][3]float64 `json:"afsAttitudeUncertainty"`
+	RateUncertainty     [3][3]float64 `json:"afsRateUncertainty"`
+	SpacecraftRate      [3]float64    `json:"afsSpacecraftRate"`
+	TangentPoint        [3]float64    `json:"afsTangentPoint"`
+}
+
+// OrbitRecord for variables in group: "PreciseOrbitEstimation"
+type OrbitRecord struct {
+	Time           time.Time
+	GnssStateJ2000 [6]float64    `json:"acsGnssStateJ2000"`
+	Uncertainty    [6][6]float64 `json:"acsNavigationUncertainty"`
+}
+
+// GnssRecord for variables in group: "TM_acGnssOps"
+type GnssRecord struct {
+	Time            time.Time
+	PropagationTime uint8   `json:"acoOnGnssPropagationTime"`
+	StateEcefVX     float32 `json:"acoOnGnssStateEcef_vx"`
+	StateEcefVY     float32 `json:"acoOnGnssStateEcef_vy"`
+	StateEcefVZ     float32 `json:"acoOnGnssStateEcef_vz"`
+	StateEcefX      float32 `json:"acoOnGnssStateEcef_x"`
+	StateEcefY      float32 `json:"acoOnGnssStateEcef_y"`
+	StateEcefZ      float32 `json:"acoOnGnssStateEcef_z"`
+	StateTime       float64 `json:"acoOnGnssStateTime"`
+}
+
+// Records collection of records
+type Records struct {
+	PowerRecords       []PowerRecord       `json:"HK_ecPowOps_1"`
+	CurrentRecords     []CurrentRecord     `json:"HK_scSysOps_1"`
+	TemperatureRecords []TemperatureRecord `json:"HK_tcThermEssential"`
+	AttitudeRecords    []AttitudeRecord    `json:"PreciseAttitudeEstimation"`
+	OrbitRecords       []OrbitRecord       `json:"PreciseOrbitEstimation"`
+	GnssRecords        []GnssRecord        `json:"TM_acGnssOps"`
+}
+
+//Write records to file
+func (r Records) Write(outputfile string) error {
+	outdata, err := json.MarshalIndent(r, "", "    ")
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(outputfile, outdata, 0644)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func getDataset(
+	filename string, groupname string, datasetname string) *hdf5.Dataset {
+
+	f, err := hdf5.OpenFile(filename, hdf5.F_ACC_RDONLY)
+	if err != nil {
+		log.Fatalln(err, filename)
+	}
+	group, err := f.OpenGroup(groupname)
+	if err != nil {
+		log.Fatalln(err, filename, groupname)
+	}
+
+	dataset, err := group.OpenDataset(datasetname)
+	if err != nil {
+		log.Fatalln(err, filename, groupname, datasetname)
+	}
+	return dataset
+}
+
+func getDatasetFloat64(
+	filename string, groupname string, datasetname string) []float64 {
+
+	dataset := getDataset(filename, groupname, datasetname)
+	npoints := dataset.Space().SimpleExtentNPoints()
+	data := make([]float64, npoints)
+	err := dataset.Read(&data)
+	if err != nil {
+		log.Fatalln(err, filename, groupname, datasetname, "data")
+	}
+	dataset.Close()
```

Comment:
`defer dataset.Close()` on row 122 could be safer (ensures closes even if error).

Reply:
thanks